llvm/lib/Transforms/Instrumentation/MemProfiler.cpp

```diff
@@ -795,6 +795,53 @@ struct AllocMatchInfo {
   bool Matched = false;
 };
 
+DenseMap<uint64_t, SmallVector<CallEdgeTy, 0>>
+memprof::extractCallsFromIR(Module &M) {
+  DenseMap<uint64_t, SmallVector<CallEdgeTy, 0>> Calls;
+
+  auto GetOffset = [](const DILocation *DIL) {
```

Comment:
Maybe move this logic into the LineLocation constructor? Then it will be fully encapsulated in that struct instead of having a bit of the logic outside and a bit inside.

Reply:
I am torn.  I understand your point, but then we would need to pass `DIL` to `LineLocation`.

Reply:
Right, any reason you are opposed to passing DIL to it?

Reply:
I see `LineLocation` as a container and as a nicer version of `std::pair` where the fields are more appropriately named than `first` and `second`.  If I put an absolute line number in there, I would like to see the same absolute line number come out, but then the absolute line number isn't really useful in our context.